### PR TITLE
Initialize lastUnsafeSegmentGenerationForGets in constructor

### PR DESCRIPTION
**Pull request by ywangd**
This PR adds initialization for `lastUnsafeSegmentGenerationForGets` in the constructor. It is meant to fix test failures for RTG during primary relocation in another PR.

### CODE REVIEW

**Comment by ywangd** [comment]
@henningandersen I repurposed this PR to initialize `lastUnsafeSegmentGenerationForGets` in the constructor of `InternalEngine`. I didn't add an unit test for it since it will be tested on the other side. Please let me know if you think otherwise. Thanks!

File: server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java [hunk 1]
```diff
@@ -290,6 +290,7 @@ public InternalEngine(EngineConfig engineConfig) {
             this.internalReaderManager = internalReaderManager;
             this.externalReaderManager = externalReaderManager;
             internalReaderManager.addListener(versionMap);
+            this.lastUnsafeSegmentGenerationForGets.set(lastCommittedSegmentInfos.getGeneration());
```
Comment (henningandersen) [inline server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java:293]: I wonder if we should init the AtomicLong here, i.e.:

```suggestion
            this.lastUnsafeSegmentGenerationForGets = new AtomicLong(lastCommittedSegmentInfos.getGeneration());
```

and remove the initialization on the field? I think it is not used until the engine is constructed.

**Review by henningandersen — APPROVED** [review]
LGTM

(left an optional suggestion).

File: server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java (hunk 1 above)
Comment (ywangd) [reply server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java]: Yeah you are right. It should be safe as you suggested. Pushed 4aab64f7bd493aac6a39419cbec0bdae06c39965

**Comment by ywangd** [comment]
@elasticmachine update branch

**Comment by ywangd** [comment]
Failure is #107699

**Pull request merged**